redbox-core/redbox/app.py

```diff
@@ -0,0 +1,100 @@
+import typing
+from langgraph.graph import StateGraph
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_core.vectorstores import VectorStoreRetriever
+from tiktoken import Encoding
+
+from redbox.chains.graph import set_route, set_state_field
+from redbox.graph.search import get_search_graph
+from redbox.models.chain import ChainState
+from redbox.models.chat import ChatRoute
+from redbox.models.settings import Settings
+from redbox.chains.components import get_all_chunks_retriever, get_parameterised_retriever, get_chat_llm, get_tokeniser
+from redbox.graph.chat import get_chat_graph, get_chat_with_docs_graph
+
+
+async def _default_callback(*args, **kwargs):
+    return None
+
+
+class Redbox:
+    FINAL_RESPONSE_TAG = "response_flag"
+    SOURCE_DOCUMENTS_TAG = "source_documents_flag"
+    ROUTE_NAME_TAG = "route_flag"
+
+    # Non keywords
+    ROUTABLE_BUILTIIN = [ChatRoute.chat, ChatRoute.chat_with_docs, ChatRoute.error_no_keyword]
+
+    # Keyword routes
+    ROUTABLE_KEYWORDS = {ChatRoute.search: "Search for an answer to the question in the document"}
+
+    def __init__(
+        self,
+        llm: BaseChatModel = None,
+        all_chunks_retriever: VectorStoreRetriever = None,
+        parameterised_retriever: VectorStoreRetriever = None,
+        tokeniser: Encoding = None,
+        env: Settings = None,
+        debug: bool = False,
```

Comment:
note-to-self: settings that define:
* llm
* all_chunks_retriever
* parameterised_retriever
* tokeniser
should be the settings that are injected into the request and not in `env = Settings`